src/components/AbuseReportForm/AbuseReportForm.tsx

```diff
@@ -44,175 +59,197 @@ const inputWrapper = css`
-        <option value="1">Personal abuse</option>
-        <option value="2">Off topic</option>
-        <option value="3">Legal issue</option>
-        <option value="4">Trolling</option>
-        <option value="5">Hate speech</option>
-        <option value="6">Offensive/Threatening language</option>
-        <option value="7">Copyright</option>
-        <option value="8">Spam</option>
-        <option value="9">Other</option>
-      </select>
-      {errors && errors.category && (
-        <span
-          className={css`
-            color: red;
-          `}
-        >
-          {errors.category}
-        </span>
-      )}
-    </div>
+    document.addEventListener("keydown", keyListener);
 
-    <div className={inputWrapper}>
-      <label htmlFor="reason">Reason (optional)</label>
-      <textarea
-        name="reason"
-        onChange={reasonOnChange}
-        value={reasonText}
-      ></textarea>
-    </div>
-
-    <div className={inputWrapper}>
-      <label htmlFor="email">Email (optional)</label>
-      <input
-        type="email"
-        name="email"
-        onChange={emailOnChange}
-        value={emailText}
-      ></input>
-    </div>
+    return () => document.removeEventListener("keydown", keyListener);
+  });
 
-    <div
-      className={css`
-        float: right;
-      `}
-    >
-      <Button onClick={submitForm} type="submit" size="small">
-        Report
-      </Button>
-    </div>
-  </form>
-);
+  const modalRef = useRef<HTMLDivElement>(null);
+  let firstElement: HTMLSelectElement | null;
+  let lastElement: HTMLButtonElement | null;
+  // We want to pull out the 1st and last elements of the form, and highlight the 1st element
+  useEffect(() => {
+    if (!modalRef.current) return;
+    firstElement = modalRef.current.querySelector('select[name="category"]');
+    lastElement = modalRef.current.querySelector(
+      'button[custom-guardian="close-modal"]'
+    );
+    firstElement && firstElement.focus();
+  }, [modalRef]);
 
-export const AbuseReportForm: React.FC<{
-  commentId: number;
-}> = ({ commentId }: Props) => {
-  const [showForm, setShowForm] = useState(false);
-  const toggleSetShowForm = () => setShowForm(!showForm);
+  // This function only gets called when a tab key has been pressed
+  // we use `e.shiftKey` internally to determin the direction of the highlighting
+  const handleTabKey = (e: KeyboardEvent) => {
+    if (!e.shiftKey && document.activeElement === lastElement) {
+      firstElement && firstElement.focus();
+      e.preventDefault();
+    }
 
-  const [errors, setErrors] = useState({});
+    if (e.shiftKey && document.activeElement === firstElement) {
+      lastElement && lastElement.focus();
+      e.preventDefault();
+    }
+  };
 
-  const [formState, setFormState] = useState({
-    categoryId: 0,
-    reason: "",
-    email: ""
+  const { register, handleSubmit, errors } = useForm({
+    defaultValues: {
+      categoryId: 0,
+      reason: "",
+      email: ""
+    }
```

Comment:
Are these used? I guess they could be part of the form lib but if not, maybe set default when destructuring below?